src/DSP/FFT/FFT.cpp

```diff
@@ -1,17 +1,22 @@
 #include "FFT.h"
 
-#include <assert.h>
+#include <cassert>
+#include <cstring>
 #include <chrono>
 
 namespace lime {
 
-FFT::FFT(uint32_t size)
-    : samplesFIFO(size * 128)
+FFT::FFT(uint8_t channelCount, uint32_t size, WindowFunctionType windowType)
+    : samplesFIFO(channelCount)
+    , channelCount(channelCount)
+    , currentWindowType(windowType)
```

Comment:
only sets the window type, but does not generate coefficients for it, and default to GenerateWindowCoefficients(WindowFunctionType::BLACKMAN_HARRIS), until next type change, resulting in GUI selection and actual value mismatch.